tests/unit/models/primo.js

```diff
@@ -0,0 +1,680 @@
+
+  });
+
+  it("primo model should exist", function() {
+    expect(primo).toBeDefined();
+  });
+
+  describe("primo model getScope method >", function() {
+    it("should retrieve the scope from a search result", function() {
+      $scope.$ctrl = {
+        parentCtrl: {
+          result: {
+            pnx: {
+              display: {
+                type: ["journal"]
+              },
+
+              addata: {
+                issn: ["0096-6762", "0028-4793"]
+              }
+            }
+          }
+        }
+      };
+
+      var scope = primo.getScope($scope);
+
+      expect(scope).toBeDefined();
+      expect(scope.result).toBeDefined();
+      expect(scope.result.pnx).toBeDefined();
+      expect(scope.result.pnx.display).toBeDefined();
+      expect(scope.result.pnx.display.type).toBeDefined();
+      expect(scope.result.pnx.addata).toBeDefined();
+      expect(scope.result.pnx.addata.issn).toBeDefined();
+
+      expect(scope.result.pnx.display.type[0]).toEqual("journal");
+      expect(scope.result.pnx.addata.issn[0]).toEqual("0096-6762");
+      expect(scope.result.pnx.addata.issn[1]).toEqual("0028-4793");
+    });
+  });
+
+  describe("primo model getResult method >", function() {
+    it("should include a result object from prmSearchResultAvailabilityLineAfterController $scope", function() {
+      var scope = {
+        result: {
+          pnx: {
+            display: {
+              type: ["journal"]
+            },
+
+            addata: {
+              issn: ["0096-6762", "0028-4793"]
+            }
+          }
+        }
+      };
+
+      var result = primo.getResult(scope);
+      expect(result).toBeDefined();
+    });
+
+    it("should include a result object from prmSearchResultThumbnailContainerAfterController $scope", function() {
+      var scope = {
+        item: {
+          pnx: {
+            display: {
+              type: ["journal"]
+            },
+
+            addata: {
+              issn: ["0096-6762", "0028-4793"]
+            }
+          }
+        }
+      };
+
+      var result = primo.getResult(scope);
+      expect(result).toBeDefined();
+    });
+  });
+
+  describe("primo model urlRewrite method >", function() {
+    it("should rewrite the public api domain", function() {
+      var url = "https://api.thirdiron.com/public/v1/libraries/XXX";
+      expect(primo.urlRewrite(url)).toEqual("https://public-api.thirdiron.com/public/v1/libraries/XXX");
+    });
+
+    it("should not rewrite the public api domain when the public-api domain already exists", function() {
+      var url = "https://public-api.thirdiron.com/public/v1/libraries/XXX";
+      expect(primo.urlRewrite(url)).toEqual("https://public-api.thirdiron.com/public/v1/libraries/XXX");
+    });
+  });
+
+  describe("primo model shouldEnhance method >", function() {
+    it("should not enhance a search result without scope data", function() {
+      var scope = {};
+
+      expect(primo.shouldEnhance(scope)).toEqual(false);
```

Comment:
Instead of just checking if this returns false, could we pass in the invalid data and confirm it hasn’t messed with it?